### ISSUE TITLE
feature: custom sort order resource list TV

### ISSUE THREAD

**Issue by j-craig** [modxcms/revolution#12894]
Currently when you create a resource list template variable there is no way to specify a sort order for the resources. It appears to use menu index as the sort which makes for something akin to pre-order traversal of the resource tree. This is fine if the resource list TV is set to only a single level of depth but if you are using it for a large section or all of your site it becomes very cumbersome.

I would propose the ability to add one or more sort fields either ascending or descending to the TV input options. I think it would be fine to have a textarea like the WHERE conditions field where you enter something like "pagetitle:DESC, longtitle:ASC" or in a JSON object. No need to get too fancy with a GUI to edit the sort fields IMO.

**Comment by pepebe** [comment]
I don't think the resource list TV will ever be able to solve everyones problem with a nice UI. I haven't used it for years. A single or multiselect list combined with an @binding gives you much more options and it is very easy to setup.

In the forum you can find a lot of examples for various use cases:

```
@SELECT pagetitle, id FROM [[+PREFIX]]site_content WHERE (parent=899 AND id<>1058) ORDER BY menuindex
```

```
  @EVAL $output = $modx->runSnippet(
    'getResources'
    ,array(
       'parents'=>'99'
      ,'tpl'=>'@INLINE [[+pagetitle]]([[+id]])==[[+id]]'
      ,'outputSeparator'=>'||'
      ,'sortby'=>'pagetitle'
      ,'sortdir'=>'ASC'
      ,'limit'=>'0'
      )
    );
    return 'choose one==||' . $output;
```

```
@SELECT CONCAT(`pagetitle`,' (',`id`, ')') AS `name`,`id` FROM `[[+PREFIX]]site_content` WHERE `published` = 1 AND `deleted` = 0
```

```
@SELECT pagetitle,contentid FROM `[[+PREFIX]]site_content` AS c JOIN `[[+PREFIX]]site_tmplvar_contentvalues` AS t ON c.id=t.contentid WHERE tmplvarid = 4 AND value = 'ja' AND template = 8
```

**Comment by alroniks** [comment]
Indeed, @-bindings (however, @eval already removed in 3.x) solves this issue very well. So closing as wontfix.
